docs/docs/training-data-format.mdx

```diff
@@ -1,355 +1,655 @@
+      - action: utter_slots_values
 ```
 
-See [Training Data Format](./training-data-format) for details on how to define entities with roles and groups in your training data.
-
-The entity object returned by the extractor will include the detected role/group label.
-
-```json
-{
-  "text": "Book a flight from Berlin to SF",
-  "intent": "book_flight",
-  "entities": [
-    {
-      "start": 19,
-      "end": 25,
-      "value": "Berlin",
-      "entity": "city",
-      "role": "departure",
-      "extractor": "DIETClassifier",
-    },
-    {
-      "start": 29,
-      "end": 31,
-      "value": "San Francisco",
-      "entity": "city",
-      "role": "destination",
-      "extractor": "DIETClassifier",
-    }
-  ]
-}
+
+#### Slots
+
+**[Slots](../core/slots)** act as the bots memory. Slots are **set** by entities or by custom actions and **referenced** by stories in `slot_was_set` steps. For example.
+
+```yaml
+    steps:
+    - intent: celebrate_bot
+    - slot_was_set: feedback_value
+      value: positive
+    - action: utter_yay
 ```
 
+This means the story requires that the current value for the `feedback_value`
+slot be `positive` for the conversation to continue as specified.
+
 :::note
-Composite entities are currently only supported by the [DIETClassifier](./components/intent-classifiers.mdx#dietclassifier) and [CRFEntityExtractor](./components/entity-extractors.mdx#crfentityextractor).
+Stories do not **set** slots. The slot must be set by an entity or custom action **before** the `slot_was_set` step.
+:::
+
+
+#### Checkpoints
+
+Checkpoints are ways to connect stories together. They can be either the first
+or the last step in a story. If they are the last step in a stories, that story
+will be connected to each other story that starts with the checkpoint of the
+same name when the model is trained. Here is an example of a story that ends
+with a checkpoint, and one that starts with the same checkpoint:
+
+```yaml
+- story: story_with_a_checkpoint_1
+  steps:
+  - intent: greet
+  - action: utter_greet
+  - checkpoint: greet_checkpoint
+
+- story: story_with_a_checkpoint_2
+  steps:
+  - checkpoint: greet_checkpoint
+  - intent: book_flight
+  - action: action_book_flight
+```
+
+Checkpoints at the beginning of stories can also be conditional on slots being set, for example:
+
+```yaml
+- story: story_with_a_conditional_checkpoint
+  steps:
+  - checkpoint: greet_checkpoint
+    # This checkpoint should only apply if slot is set to the specified value
+    slots: 
+    - context_scenario: holiday
+    - holiday_name: thanksgiving
+  - intent: greet
+  - action: utter_greet_thanksgiving
+```
+
+
+Checkpoints can help simplify your training data and reduce redundancy in it,
+but **do not overuse them**. Using lots of checkpoints can quickly make your
+stories hard to understand. It makes sense to use them if a sequence of steps
+is repeated often in different stories, but stories without checkpoints
+are easier to read and write. 
```

Comment:
isn't this pretty much the use case for rules?

Reply:
It does seem like it. I'm actually not sure of a concrete example for checkpoints 🤔

Reply:
I guess the idea is if you have like a feedback form flow, you could use a checkpoint to make sure that after each of your use case flows, you go right into the feedback flow (without having to have the full feedback flow written down everywhere, making it easier if you change the feedback flow)? I feel like that's maybe something that'd be easier with checkpoints than rules... 🤔 

in other words maybe it's good for a sequence of steps that's repeated often, but started by the bot and not the user? but i also have no idea of rules and if you can accomplish this with them (why am i reviewing these docs again? lmao)

Reply:
That's true actually, like a sequence of follow up actions?

Reply:
Hey I dunno what's up with rules yet either 😂

Reply:
![tenor](https://user-images.githubusercontent.com/22333512/90113627-edcdd300-dd51-11ea-973c-12e64740263f.gif)

Reply:
I think let's wait for clarification on rules & checkpoints and try again for a better explanation in the next round?

Reply:
Sounds good to me